docs/Deprecating-Disabling-and-Removing-Casks.md

```diff
@@ -0,0 +1,99 @@
+# Deprecating, Disabling and Removing Casks
+
+There are many reasons why casks may be deprecated, disabled or removed. This document explains the differences between each method as well as explaining when one method should be used over another.
+
+## Overview
+
+These general rules of thumb can be followed:
+
+- `deprecate!` should be used for casks that _should_ no longer be used.
+- `disable!` should be used for casks that _cannot_ be used.
+- Casks that are no longer acceptable in `homebrew/cask` or have been disabled for over a year _should_ be removed.
+
+## Deprecation
+
+If a user attempts to install a deprecated cask, they will be shown a warning message but the install will proceed.
+
+A cask should be deprecated to indicate to users that the cask should not be used and will be disabled in the future. Deprecated casks should continue to be maintained by the Homebrew maintainers if they continue to be installable. If this is not possible, they should be immediately disabled.
+
+The most common reasons for deprecation are when the upstream project is deprecated, unmaintained or archived.
+
+Casks should only be be deprecated if at least one of the following are true:
+
+- the cask does not install on any of our supported macOS versions
+- the cask has outstanding CVEs
+- the cask has [zero installs in the last 90 days](https://formulae.brew.sh/analytics/cask-install/90d/)
+
+To deprecate a cask, add a `deprecate!` call. This call should include a deprecation date in the ISO 8601 format and a deprecation reason:
+
+```ruby
+deprecate! date: "YYYY-MM-DD", because: :reason
+```
+
+The `date` parameter should be set to the date that the project or version became (or will become) deprecated. If there is no clear date but the cask needs to be deprecated, use today's date. If the `date` parameter is set to a date in the future, the cask will not become deprecated until that date. This can be useful if the upstream developers have indicated a date when the project or version will stop being supported. Do not backdate the `date` parameter as it causes confusion for users and maintainers.
```

Comment:
```suggestion
The `date` parameter should be set to the date that the deprecation period should begin, which is usually today's date. If the `date` parameter is set to a date in the future, the cask will not become deprecated until that date. This can be useful if the upstream developers have indicated a date when the project or version will stop being supported. Do not backdate the `date` parameter as it causes confusion for users and maintainers.
```